src/wallet/wallet2.cpp

```diff
@@ -14114,11 +14096,12 @@ void wallet2::stop_background_sync(const epee::wipeable_string &wallet_password,
 
   // Set the plaintext spend key
   m_account.set_spend_key(recovered_spend_key);
+  const bool should_reencrypt_keys = m_ask_password == AskPasswordToDecrypt && !m_unattended && !m_watch_only;
```

Comment:
This predicate seems common enough that you might consider introducing an inline for it.

Reply:
I agree. I did make a shorthand in the FCMP++/Carrot downstream branch, maybe I should upstream it in #9860

Reply:
Okay done